src/map.h

```diff
@@ -307,16 +307,29 @@ class MMVManip : public VoxelManipulator
 	MMVManip(Map *map);
 	virtual ~MMVManip() = default;
 
-	virtual void clear()
-	{
-		VoxelManipulator::clear();
-		m_loaded_blocks.clear();
-	}
-
+	/*
+		Loads specified area from map and *adds* it to the area already
+		contained in the VManip.
+	*/
 	void initialEmerge(v3s16 blockpos_min, v3s16 blockpos_max,
 		bool load_if_inexistent = true);
 
-	// This is much faster with big chunks of generated data
+	/**
+		Uses the flags array to determine which blocks the VManip covers,
+		and for which of them we have any data.
+		@warning requires VManip area to be block-aligned
```

Comment:
Otherwise it'll throw an exception. That's not very nice. But I do not see an easy way out.